src/feed/api.ts

```diff
@@ -81,6 +83,15 @@ export async function writeFeedDataRaw(
   const topicHash = bmtHashString(topic)
   const id = getId(topicHash, epoch.time, epoch.level)
   const socWriter = connection.bee.makeSOCWriter(privateKey)
-
-  return socWriter.upload(connection.postageBatchId, id, data)
+  try {
+    return await socWriter.upload(connection.postageBatchId, id, data, {
+      pin: true,
+    })
+  } catch (e) {
+    if (errorStartWith(e, CHUNK_ALREADY_EXISTS_ERROR)) {
+      return getBmtDataAddress(data)
```

Comment:
the BMT address of the `data` is not identical with the SOC address. the SOC address is the hash of the uploader's ethereum address and the identifier.
Also, a new feed update must be made with the already uploaded data's reference.